### ISSUE TITLE
SIGSEGV Encountered after Reward Call

### ISSUE THREAD

**Issue by GistKen** [livepeer/go-livepeer#1964]
**Describe the bug**
SIGSEGV Encountered after Reward Call

**To Reproduce**
With the automatic reward call enabled, send a reward call. Unclear what is specific about this configuration that resulted in  SIGSEGV

**Expected behavior**
Livepeer encountered a invalid memory address or nil pointer dereference during the timeout period after attempting to call a reward. The reward call occurred at 6:48:13 with the SIGSEGV at 06:53:13.
Max Gas price was set to 2500000 GWei. Using the alchemy service

**Screenshots**
```
Jul 27 06:48:13 Cosmos livepeer[20146]: ***************************************************************************
Jul 27 06:53:13 Cosmos livepeer[20146]: I0727 06:53:13.466748   20146 transactionManager.go:186]
Jul 27 06:53:13 Cosmos livepeer[20146]: ******************************Eth Transaction******************************
Jul 27 06:53:13 Cosmos livepeer[20146]: Replacement transaction: "rewardWithHint".  Gas Price: 28600000010
Jul 27 06:53:13 Cosmos livepeer[20146]: Transaction Failed: <nil>
Jul 27 06:53:13 Cosmos livepeer[20146]: ***************************************************************************
Jul 27 06:58:13 Cosmos livepeer[20146]: panic: runtime error: invalid memory address or nil pointer dereference
Jul 27 06:58:13 Cosmos livepeer[20146]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1054eed]
Jul 27 06:58:13 Cosmos livepeer[20146]: goroutine 135 [running]:
Jul 27 06:58:13 Cosmos livepeer[20146]: github.com/livepeer/go-livepeer/eth.(*client).CheckTx(0xc000d5c000, 0xc000909200, 0x0, 0x0)
Jul 27 06:58:13 Cosmos livepeer[20146]: #011/home/runner/work/go-livepeer/go-livepeer/eth/client.go:878 +0x1ed
Jul 27 06:58:13 Cosmos livepeer[20146]: github.com/livepeer/go-livepeer/eth.(*RewardService).tryReward(0xc000279770, 0xc000086e74, 0x3)
Jul 27 06:58:13 Cosmos livepeer[20146]: #011/home/runner/work/go-livepeer/go-livepeer/eth/rewardservice.go:94 +0x171
Jul 27 06:58:13 Cosmos livepeer[20146]: github.com/livepeer/go-livepeer/eth.(*RewardService).Start(0xc000279770, 0x2041cc0, 0xc00019c000, 0x0, 0x0)
Jul 27 06:58:13 Cosmos livepeer[20146]: #011/home/runner/work/go-livepeer/go-livepeer/eth/rewardservice.go:54 +0x2fb
Jul 27 06:58:13 Cosmos livepeer[20146]: main.main.func4(0xc000279770, 0x2041cc0, 0xc00019c000, 0xc00069df80)
Jul 27 06:58:13 Cosmos livepeer[20146]: #011/home/runner/work/go-livepeer/go-livepeer/cmd/livepeer/livepeer.go:701 +0x3f
Jul 27 06:58:13 Cosmos livepeer[20146]: created by main.main
Jul 27 06:58:13 Cosmos livepeer[20146]: #011/home/runner/work/go-livepeer/go-livepeer/cmd/livepeer/livepeer.go:700 +0x738c
Jul 27 06:58:13 Cosmos livepeer[20250]: I0727 06:58:13.476952   20250 ot_rpc.go:120] End of stream receive cycle because of err="rpc error: code = Unavailable desc = transport is closing", waiting for running transcode jobs to complete
Jul 27 06:58:13 Cosmos livepeer[20250]: I0727 06:58:13.477035   20250 ot_rpc.go:53] Unregistering transcoder: rpc error: code = Unavailable desc = transport is closing
Jul 27 06:58:13 Cosmos systemd[1]: livepeer_orchestrator.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Jul 27 06:58:13 Cosmos systemd[1]: livepeer_orchestrator.service: Failed with result 'exit-code'.
```

**Desktop (please complete the following information):**
 Node's version 0.5.18-93dfb024 
 Node's GO runtime version go1.15.5 |
 Node's architecture  amd64

**Comment by yondonfu** [comment]
Thanks for the bug report!

A note for the person working on a fix for this:

The node version included in the bug report is 0.5.18-93dfb024 which references a recent commit on [this branch](https://github.com/livepeer/go-livepeer/commits/yf/pixels-o). While that branch is a few commits behind master, none of the missing commits are related to the area of the codebase where this bug is being triggered so the bug should be present on master as well.

**Comment by Franck-UltimaRatio** [comment]
For your information, I encountered the same problem after 3 reward calls which took too long.
